modyn/evaluator/internal/metrics/abstract_evaluation_metric.py

```diff
@@ -0,0 +1,46 @@
+from abc import ABC, abstractmethod
+
+import torch
+
+
+class AbstractEvaluationMetric(ABC):
+    """
+    This abstract class is used to represent an evaluation metric which can be used to evaluate a trained model.
+    """
+
+    def __init__(self, progressive: bool):
+        """
+        Abstract class for all evaluation metrics.
+        Currently we only allow progressive metrics i.e metrics where the final evaluation value can be calculated
+        on-the-fly without having the model output from all batches.
+
+        Args:
+            progressive: whether the metrics is progressive.
+        """
+        self.progressive = progressive
+
+        if not self.progressive:
+            raise NotImplementedError("Only progressive metrics are allowed.")
+
+    @abstractmethod
+    def evaluate_batch(self, y_true: torch.tensor, y_pred: torch.tensor, batch_size: int) -> None:
```

Comment:
maybe rename batch_evaluated_callback? not sure WDYT?

Reply:
Naming is hard^^
Yes, I think this makes sense. It's a function that is called whenever we can make progress in our metric calculation (meaning that model outputs and labels are ready to be processed for one batch).